modules/ppcp-settings/resources/js/Components/ReusableComponents/ConnectionInfo.js

```diff
@@ -1,54 +1,58 @@
+    const [connectionData, setConnectionData] = useState(getDefaultConnectionStatusData(merchant));
 
-	if ( connectionData.showAllData ) {
-		toggleStatusClassName.push(
-			'ppcp-r-connection-status__status-toggle--toggled'
-		);
-	}
+    useEffect(() => {
+        if (merchant) {
+            setConnectionData(getDefaultConnectionStatusData(merchant));
+        }
+    }, [merchant]);
 
 	return (
-		<div className="ppcp-r-connection-status__data">
-			<div className="ppcp-r-connection-status__status-row ppcp-r-connection-status__status-row--first">
-				<span className="ppcp-r-connection-status__status-label">
-					{ __( 'Merchant ID', 'woocommerce-paypal-payments' ) }
-				</span>
-				<span className="ppcp-r-connection-status__status-value">
-					{ connectionData.merchantId }
-				</span>
-			</div>
-			<div className="ppcp-r-connection-status__status-row">
-				<span className="ppcp-r-connection-status__status-label">
-					{ __( 'Email address', 'woocommerce-paypal-payments' ) }
-				</span>
-				<span className="ppcp-r-connection-status__status-value">
-					{ connectionData.email }
-				</span>
-			</div>
-			<div className="ppcp-r-connection-status__status-row">
-				<span className="ppcp-r-connection-status__status-label">
-					{ __( 'Client ID', 'woocommerce-paypal-payments' ) }
-				</span>
-				<span className="ppcp-r-connection-status__status-value">
-					{ connectionData.clientId }
-				</span>
-			</div>
-		</div>
-	);
+        <div className="ppcp-r-connection-status__data">
+            {renderStatusRow(
+                __('Merchant ID', 'woocommerce-paypal-payments'),
+                connectionData.merchantId
+            )}
+            {renderStatusRow(
+                __('Email address', 'woocommerce-paypal-payments'),
+                connectionData.email
+            )}
+            {renderStatusRow(
+                __('Client ID', 'woocommerce-paypal-payments'),
+                connectionData.clientId
+            )}
+        </div>
+    );
 };
 export default ConnectionInfo;
+export const getDefaultConnectionStatusData = (merchant = null) => {
+    if (!merchant) {
+        const contextMerchant = CommonHooks.useMerchantInfo()?.merchant || {};
+        return {
+            connectionStatus: contextMerchant.isConnected || false,
+            email: contextMerchant.email || '',
+            merchantId: contextMerchant.id || '',
+            clientId: contextMerchant.clientId || '',
+        };
+    }
 
-export const connectionStatusDataDefault = {
-	connectionStatus: true,
-	showAllData: false,
-	email: 'bt_us@woocommerce.com',
-	merchantId: 'AT45V2DGMKLRY',
-	clientId: 'BAARTJLxtUNN4d2GMB6Eut3suMDYad72xQA-FntdIFuJ6FmFJITxAY8',
+    return {
+        connectionStatus: merchant.isConnected || false,
+        email: merchant.email || '',
+        merchantId: merchant.id || '',
+        clientId: merchant.clientId || '',
+    };
 };
+
+const renderStatusRow = (label, value) => (
```

Comment:
Suggestion: I would turn this into a Component instead of a function

```js
const StatusRow = ({label, value}) => (
    // no change in the content
}

...

const ConnectionInfo = () => {
    // ...
    <StatusRow
        label={ __('Email address', 'woocommerce-paypal-payments') }
        value={ merchant.email }
    />
    // ...
}
```